airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -241,6 +249,40 @@ public List<TableInfo<CommonField<JDBCType>>> discoverInternal(final JdbcDatabas
     }
   }
 
+  @Override
+  protected boolean verifyCursorColumnValues(final JdbcDatabase database, final String schema, final String tableName, final String columnName)
+      throws SQLException {
+
+    boolean nullValExist = false;
+    final String resultColName = "nullValue";
+    final String descQuery = String.format(DESCRIBE_TABLE_QUERY, tableName);
+    final Optional<JsonNode> field = database.bufferedResultSetQuery(conn -> conn.createStatement()
+        .executeQuery(descQuery),
+        resultSet -> JdbcUtils.getDefaultSourceOperations().rowToJson(resultSet))
+        .stream()
+        .filter(x -> x.get("TABLE_OWNER").asText().equals(schema))
+        .filter(x -> x.get("COLUMN_NAME").asText().equalsIgnoreCase(columnName))
+        .findFirst();
+    if (field.isPresent()) {
+      final JsonNode jsonNode = field.get();
+      final JsonNode isNullable = jsonNode.get("IS_NULLABLE");
+      if (isNullable != null) {
+        if (isNullable.asText().equalsIgnoreCase("YES")) {
+          final String query = String.format(NULL_CURSOR_VALUE_WITH_SCHEMA_QUERY,
+              schema, tableName, columnName, resultColName);
+
+          LOGGER.debug("null value query: {}", query);
+          final List<JsonNode> jsonNodes = database.bufferedResultSetQuery(conn -> conn.createStatement().executeQuery(query),
+              resultSet -> JdbcUtils.getDefaultSourceOperations().rowToJson(resultSet));
+          Preconditions.checkState(jsonNodes.size() == 1);
+          nullValExist = jsonNodes.get(0).get(resultColName).booleanValue();
+          LOGGER.debug("null value exist: {}", nullValExist);
+        }
+      }
+    }
+    return !nullValExist;
```

Comment:
return true here as well, with a similar comment (will enable after we have sent comms to users this affects)